dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/report/TrackerImportReport.java

```diff
@@ -92,4 +100,111 @@
      */
     @JsonProperty
     String message;
+
+    /**
+     * Factory method to use in case one or more Validation errors are present
+     * in the {@link TrackerValidationReport} and the Import process needs to
+     * exit without attempting persistence.
+     *
+     * Import statistics are calculated assuming that the persistence stage was
+     * never attempted, therefore all bundle objects were ignored.
+     *
+     * @param validationReport The validation report
+     * @param timingsStats The timing stats
+     * @param bundleSize The sum of all bundle objects
+     *
+     */
+    public static TrackerImportReport withValidationErrors(
```

Comment:
just as a general note: a method like this where at least from the name we are adding something maybe is best done using the `toBuilder()` you can get with `@Builder( toBuilder = true )`. That way you can not just start from a certain point but generally add to a complex object using the builder to construct a new immutable object with some things changed.

Reply:
will look into it, thank you!